### PR TITLE
[vcf-combiner] Docs update and restructuring.

### PR DESCRIPTION
**Pull request by tpoterba**
* Made vcf_combiner a module in experimental. Will probably
  move this out of experimental in the future.
* Fleshed out docs, and moved associated functions into
  the same file.
* Changed run_combiner to require sample_names only if
  the header argument is defined. Will change the command-line
  script to accept a file with one column of paths, too.

~Stacked on #8396~

### CODE REVIEW

File: hail/python/hail/experimental/vcf_combiner/__init__.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+from .vcf_combiner import run_combiner, transform_gvcf, transform_one, parse_as_fields
+from .sparse_split_multi import sparse_split_multi
+from .sparse_mt_utils import lgt_to_gt
+from .densify import densify
+
+__all__ = [
+    'run_combiner',
+    'transform_gvcf',
+    'transform_one',
```
Comment (chrisvittal) [inline hail/python/hail/experimental/vcf_combiner/__init__.py:9]: Need to add `combine_gvcfs` here too.

Comment (chrisvittal) [inline hail/python/hail/experimental/vcf_combiner/__init__.py:9]: `combine_gvcfs` should be exported.

Comment (tpoterba) [reply hail/python/hail/experimental/vcf_combiner/__init__.py:9]: combine_gvcfs shouldn't be one of the user-exposed functions -- it should be totally internal to `vcf_combiner.py`. I'll build a replacement that can do incremental merges, but that will come in a separate PR.

File: hail/python/hail/docs/experimental/vcf_combiner.rst [hunk 2]
```diff
@@ -1,83 +1,190 @@
...
+    equivalent GT: 0/0
+    equivalent AD: [8, 0, 0]
+    equivalent PL: [0, 21, 42*, 21, 42*, 42*]
+
+    Sample1 (called CAT/TAT)
+    -------------------------
+    DP: 9
+    GQ: 77
+    LA: [0, 2]
+    LGT: 0/2
```
[the remark this answers is not in the document]
Comment (tpoterba) [reply hail/python/hail/docs/experimental/vcf_combiner.rst:147]: why 0/1?

[the remark this answers is not in the document]
Comment (chrisvittal) [reply hail/python/hail/docs/experimental/vcf_combiner.rst:147]: I didn't explain. My bad. This is wrong. `lgt_to_gt` will fail if `LGT` is 0/2. Similarly `unphased_diploid_gt_index` will be wrong for `LPL` here.

File: hail/python/hail/experimental/vcf_combiner/__init__.py (hunk 1 above)
Comment (chrisvittal) [reply hail/python/hail/experimental/vcf_combiner/__init__.py]: That's reasonable. The config adds a lot of the functionality that wasn't being provided to users regarding merge plans. LGTM.

File: hail/python/hail/docs/experimental/vcf_combiner.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (tpoterba) [reply hail/python/hail/docs/experimental/vcf_combiner.rst:147]: oh, yes! I miscopied this from the example I was looking at. Thanks for clarifying.

File: hail/python/hail/experimental/vcf_combiner/vcf_combiner.py [hunk 3]
```diff
@@ -159,15 +167,17 @@ def transform_gvcf(mt, info_to_keep=[]) -> Table:
     transform_row = _transform_rows_function_map[mt.row.dtype]
     return Table(TableMapRows(mt._tir, Apply(transform_row._name, transform_row._ret_type, TopLevelReference('row'))))
 
+
 def transform_one(mt, info_to_keep=[]) -> Table:
```
[the remark this answers is not in the document]
Comment (tpoterba) [reply hail/python/hail/experimental/vcf_combiner/vcf_combiner.py:171]: will do that in a followup, this is big enough

[the remark this answers is not in the document]
Comment (chrisvittal) [reply hail/python/hail/experimental/vcf_combiner/vcf_combiner.py:171]: Ok. As long as we get rid of it.

**Pull request merged**